shared/src/test/diff/nu/NoThisCtor.mls

```diff
@@ -64,8 +64,8 @@ class Foo4() extends Foo() {
   let bb = bar(0) // call `this` indirectly
 }
 //│ ╔══[ERROR] Cannot access `this` while initializing field bb
-//│ ║  l.64: 	  let bb = bar(0) // call `this` indirectly
-//│ ╙──      	      ^^^^^^^^^^^
+//│ ║  l.63: 	  fun bar(x) = this.foo + x // ok
```

Comment:
Hmm this case shows that we should still sometimes show the place where the problem arises (here the let binding location that was previously shown) in addition to the place where `this` occurs. We should omit the former when the latter is included in it.

Reply:
Is this one better:
```ts
:e
class Foo4() extends Foo() {
  val foo: Int
  virtual val foo = 4
  fun bar(x) = this.foo + x // ok
  let bb = bar(0) // call `this` indirectly
}
//│ ╔══[ERROR] Cannot access `this` while initializing field bb
//│ ║  l.67: 	  let bb = bar(0) // call `this` indirectly
//│ ║        	      ^^^^^^^^^^^
//│ ╟── The access to `this` is here
//│ ║  l.66: 	  fun bar(x) = this.foo + x // ok
//│ ╙──      	                   ^^^^
//│ class Foo4() extends Foo {
//│   fun bar: Int -> Int
//│   let bb: Int
//│   let foo: Int
//│ }
```

Reply:
Much better! But why is `.foo` underlined instead of `this`? 🤔

Reply:
I replaced the boolean flag with the `Opt[Var]` and now we can store the location of `this`.